### ISSUE TITLE
Error response from daemon: Error looking up volume plugin rbd

### ISSUE THREAD

**Issue by aisensiy** [yp-engineering/rbd-docker-plugin#30]
When I run command:

```
docker run --volume-driver=rbd --volume testimage:/mnt/foo -it busybox sh
```

It got:

```
Error response from daemon: Error looking up volume plugin rbd: Plugin Error: Plugin.Activate, 400 Bad Request: malformed Host header
```

And the log showed nothing.

But `curl-unix-socket` test works fine:

```
root@ip-172-31-10-119:/home/ubuntu# curl-unix-socket -v -X POST unix:///run/docker/plugins/rbd.sock:/Plugin.Activate
> POST /Plugin.Activate HTTP/1.1
> Socket: /run/docker/plugins/rbd.sock
> Content-Length: 0
>
< HTTP/1.1 200 OK
< Content-Type: application/vnd.docker.plugins.v1.1+json
< Date: Fri, 29 Apr 2016 07:40:57 GMT
< Content-Length: 33
{"Implements": ["VolumeDriver"]}

root@ip-172-31-10-119:/home/ubuntu# curl-unix-socket -v -X POST -d '{"Name": "testimage"}' unix:///run/docker/plugins/rbd.sock:/VolumeDriver.Create
> POST /VolumeDriver.Create HTTP/1.1
> Socket: /run/docker/plugins/rbd.sock
> Content-Length: 21
>
< HTTP/1.1 200 OK
< Content-Type: application/vnd.docker.plugins.v1.1+json
< Date: Fri, 29 Apr 2016 07:41:39 GMT
< Content-Length: 56
{"Mountpoint":"","Err":"","Volumes":null,"Volume":null}
```

**Comment by aisensiy** [comment]
I found that it seems like because of the go version. It failed in 1.6 but when I try 1.5.1 and 1.5.4 there is another error message:

```
import cycle not allowed
package github.com/yp-engineering/rbd-docker-plugin
    imports errors
    imports runtime
    imports runtime/internal/atomic
    imports runtime
```

when 

```
go get -u github.com/yp-engineering/rbd-docker-plugin
```

**Comment by porcupie** [comment]
Yes I have confirmed this is a golang version issue - go 1.6 is much stricter with host header for HTTP over socket.  Docker itself is not sending correct HOST header, so the plugin action fails when compiled with go 1.6.

For now, the fix is to recompile plugin with golang 1.5. In future versions of Docker daemon, they will send correct HOST header so we can revisit/update then.
- official docker tracking issue: https://github.com/docker/docker/issues/20865 
- similar issue in rexray plugin: https://github.com/emccode/rexray/issues/317 
- golang change in 1.6 enforces Host header: https://groups.google.com/forum/#!topic/golang-codereviews/CPeFXzMJRzI

**Comment by aisensiy** [comment]
Thanks for your reply. Now I use the docker way to build the plugin and everything goes will.
